src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java

```diff
@@ -33,37 +41,137 @@
+		Skript.registerExpression(ExprExplodedBlocks.class, Block.class, ExpressionType.SIMPLE, "[the] exploded blocks");
 	}
 	
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
-		if (!getParser().isCurrentEvent(EntityExplodeEvent.class)) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(BlockExplodeEvent.class, EntityExplodeEvent.class)) {
 			Skript.error("Exploded blocks can only be retrieved from an explode event.");
 			return false;
 		}
 		return true;
 	}
 	
-	@Nullable
 	@Override
+	@Nullable
 	protected Block[] get(Event e) {
-		List<Block> blockList = ((EntityExplodeEvent) e).blockList();
-		return blockList.toArray(new Block[blockList.size()]);
+		return getBlocks(e).toArray(new Block[0]);
+	}
+
+	@Override
+	@Nullable
+	public Iterator<? extends Block> iterator(Event e) {
+		List<Block> blocks = getBlocks(e);
+		return new ArrayList(blocks).iterator(); // Prevent ConcurrentModificationException
+	}
+
+	@Override
+	public @Nullable <R> Expression<? extends R> getConvertedExpression(Class<R>... to) {
+		return null;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(final ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case SET:
+			case DELETE:
+				return CollectionUtils.array(Block[].class);
+			case REMOVE:
+			case REMOVE_ALL:
+				return CollectionUtils.array(Block[].class, ItemType[].class);
+			case RESET:
+			default:
+				return null;
+		}
+	}
+	
+	@Override
+	public void change(Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+		List<Block> blocks = getBlocks(e);
+
+		switch (mode) {
+			case DELETE:
+			case SET:
+			case ADD:
+				if (mode != ChangeMode.ADD)
+					blocks.clear();
+
+				if (mode != ChangeMode.DELETE) {
+					assert delta != null;
+					for (Object o : delta) {
+						if (((Block) o).getType() != Material.AIR) // Performance
+							blocks.add((Block) o);
+					}
+				}
+				break;
+
+			case REMOVE:
+			case REMOVE_ALL: // ItemType here will allow `remove [all] %itemtypes% from exploded blocks` to remove all/specific amount of specific block type that matches an itemtype, a shortcut for looping
+				assert delta != null;
+				int amountRemoved = 0; // Remove X alias
+
+				for (Object o : delta) {
+					if (o instanceof Block) {
+						blocks.remove((Block) o); // no need to check whether they are equals, since blocks are only passed by reference
+					} else if (o instanceof ItemType) {
+						ItemType item = ((ItemType) o);
+						int amount = item.getInternalAmount(); // -1 when using 'all' or 'any' of alias
+
+						if (amount == -1 || mode == ChangeMode.REMOVE_ALL) { // all %itemtype% OR REMOVE_ALL
+							blocks.removeIf(item::isOfType);
+							break;
+						} else if (amount > 0)
+							for (Block b : new ArrayList<>(blocks)) {
+								if (amountRemoved >= amount)
```

Comment:
What would happen when you use `remove 5 dirt and 2 stone from exploded blocks`? Looks like it'll not remove any stone, but could be wrong